lib/nerves_ssh.ex

```diff
@@ -26,18 +26,24 @@ defmodule NervesSSH do
     defstruct opts: [], sshd: nil, sshd_ref: nil
   end
 
+  defp via_name(name), do: {:via, Registry, {NervesSSH.Registry, name}}
+
   @doc false
-  @spec start_link(Options.t()) :: GenServer.on_start()
+  @spec start_link(Options.t() | {atom(), Options.t()}) :: GenServer.on_start()
+  def start_link({name, %Options{} = opts}) do
+    GenServer.start_link(__MODULE__, %{opts | name: name}, name: via_name(name))
+  end
+
   def start_link(%Options{} = opts) do
-    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+    GenServer.start_link(__MODULE__, %{opts | name: :default}, name: via_name(:default))
```

Comment:
```suggestion
GenServer.start_link(__MODULE__, opts, name: via_name(opts.name))
```
Let's expect options to handle the default name